engine/src/main/scala/cromwell/webservice/routes/wes/WesRunRoutes.scala

```diff
@@ -20,42 +27,49 @@ trait WesRunRoutes {
+      concat(
         pathPrefix("runs") {
-          pathEnd {
-            get {
-              parameters(("page_size".as[Int].?, "page_token".?)) { (pageSize, pageToken) =>
-                completeCromwellResponse(listRuns(pageSize, pageToken, serviceRegistryActor))
-              }
+          get {
+            parameters(("page_size".as[Int].?, "page_token".?)) { (pageSize, pageToken) =>
+              WesRunRoutes.completeCromwellResponse(WesRunRoutes.listRuns(pageSize, pageToken, serviceRegistryActor))
             }
           }
+          concat(
+            path(Segment) { workflowId =>
+              get {
+                completeCromwellResponse(runLog(workflowId, (w: WorkflowId) => GetSingleWorkflowMetadataAction(w, None, None, expandSubWorkflows = false), serviceRegistryActor))
+              }
+            }
+          )
         }
+      )
     }
 }
 
 object WesRunRoutes {
 
-import akka.util.Timeout
-import scala.concurrent.duration.FiniteDuration
-import net.ceedubs.ficus.Ficus._
-
-implicit lazy val duration: FiniteDuration = ConfigFactory.load().as[FiniteDuration]("akka.http.server.request-timeout")
-implicit lazy val timeout: Timeout = duration
+  implicit lazy val duration: FiniteDuration = ConfigFactory.load().as[FiniteDuration]("akka.http.server.request-timeout")
+  implicit lazy val timeout: Timeout = duration
 
-def completeCromwellResponse(future: => Future[WesResponse]): Route = {
-
-  import WesResponseJsonSupport.WesResponseErrorFormat
-  import cromwell.webservice.routes.wes.WesResponseJsonSupport.WesResponseFormat
-  import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+  def completeCromwellResponse(future: => Future[WesResponse]): Route = {
+    onComplete(future) {
+      case Success(response: WesResponse) => complete(response)
+      case Failure(e) => complete(WesErrorResponse(e.getMessage, StatusCodes.InternalServerError.intValue))
+    }
+  }
 
-  onComplete(future) {
-    case Success(response: WesResponse) => complete(response)
-    case Failure(e) => complete(WesErrorResponse(e.getMessage, StatusCodes.InternalServerError.intValue))
+  def listRuns(pageSize: Option[Int], pageToken: Option[String], serviceRegistryActor: ActorRef): Future[WesResponse] = {
+    // FIXME: to handle - page_size, page_token
+    // FIXME: How to handle next_page_token in response?
+    metadataQueryRequest(Seq.empty[(String, String)], serviceRegistryActor).map(RunListResponse.fromMetadataQueryResponse)
   }
-}
 
-def listRuns(pageSize: Option[Int], pageToken: Option[String], serviceRegistryActor: ActorRef): Future[WesResponse] = {
-  // FIXME: to handle - page_size, page_token
-  // FIXME: How to handle next_page_token in response?
-  metadataQueryRequest(Seq.empty[(String, String)], serviceRegistryActor).map(RunListResponse.fromMetadataQueryResponse)
+  def runLog(workflowId: String, request: WorkflowId => BuildMetadataJsonAction, serviceRegistryActor: ActorRef): Future[WesResponse] = {
```

Comment:
OOI, why is the `request` a function? Couldn't you just take in the workflow ID here and build the `GetSingleWorkflowMetadataAction` directly within this function?

Reply:
Investigating...

Reply:
I don't have a good answer for why, but we copied this from existing code at my suggestion

https://github.com/broadinstitute/cromwell/blob/develop/engine/src/main/scala/cromwell/webservice/routes/MetadataRouteSupport.scala#L151